core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -445,7 +460,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
             // reading a garbage file is safe, but we would log an error which can be scary to
             // the end-user.
             !entry.getPath().getName().startsWith(".") &&
-            !isBlacklisted(entry.getPath)
+            !isBlacklisted(entry.getPath) &&
+            !isProcessing(entry.getPath)
```

Comment:
Filtering a entry currently in processing list, will skip updating lastProcessed field below. In the end of checkForLogs, the information not update newest lastProcessed will be deleted. Now checkForLogs is not blocked by tasks in each iteration, will the entries in processing list be deleted because their lastProcessed are not updated?

Reply:
It seems that only the path that was in listing and  not been scanned this time will be removed.
But the paths in `processing` are not contained in `listing`, so would not be deleted.

Reply:
As @vanzin asked, could it be in the listing because of previous scan?

Reply:
I think it only happens when the parsing has been completed and the finally block has not been invoked.
But, it should has no impaction. 

https://github.com/turboFei/spark/blob/66f97639cf5ab47132f624677c25a35a08f2bf4c/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L690-L693